docker-entrypoint.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e
+
```

Comment:
```suggestion
# There are 2 checks in this bash script.
# the first check checks if the first character in the first argument is - like so:
#
# 1=foo
# echo $1     # output: foo
# echo ${1#-} # output: foo
#
# 1=--foo       
# echo $1     # output: --foo
# echo ${1#-} # output: -foo
# the second check checks if the first argument is a command that exists in the system.
# the flag -z check if there is an output. if the command is found on the system command -v "${1}" will return its # # path (!true) if not it will return nothing (!false).

# TLDR:
# IF (is the first character in the first argument is -) OR (NOT(is the first argument a command)) THEN run dockly
# 
# WHY: we do this to enable users to run commands on the container such as opening a shell
```

Reply:
Added comments to this script for our future selves